### PR TITLE
Table visual improvements

### PR DESCRIPTION
**Pull request by kdoh**
screen shot 2018-10-09 at 2 32 44 pm

Some work to improve the out of the box visual vibes of tables. Main things include tightening up the typography, type color and spacing of cells and headings. 

TODOs
- [x] Figure out if we're going with uppercase table headings
- [X] Handle currency in headers once `series-settings` is merged
- [x] Add ID bubbles to dash tables
- [x] ~~Fix header click table blink~~ (apparently this is not a regression specific to this branch)

### CODE REVIEW

**Comment by tlrobinson** [comment]
I'm liking that better.

One idea for a small tweak to the ID bubbles: give them a `min-width: 35px` (or so) and `text-align: center` so single digit IDs are consistently oval rather than circles.

screenshot 2018-10-08 18 13 04

**Comment by kdoh** [comment]
I think this is pretty much to a point where I'll need a little assistance from @tlrobinson to get it over the finish line.

Biggest things remaining:

- `isSorted` seems to not be being properly reflected as a class on column headers which is preventing me from adding a treatment to show which is the current column being sorted by (at least it seems that way). 
- There's a strange flicker effect when you click on headings where the text of the raw table goes from formatted to all black. I've tried to figure out what it is but I'm not making much headway.
- We'll want to add a setting to turn the "token" formatting on / off for PK and FK fields.
	At this point I think we'll want to have them on by default especially if someone has done a remapping from ID to name but I don't want to force them on people if they don't need them for exploration. I'm not sure what the best title is yet but as long as we have a bit to flip that'll let me start working on that.
- I think the initial column widths logic might need some tweaking for columns that have the tokenization treatment, otherwise they ellipsify initially around 100 or so instead of 1000 or so in the current state of affairs.

**Comment by mazameli** [comment]
Are we thinking this PR will include changes to dashboard tables?

screen shot 2018-10-09 at 2 43 57 pm

**Comment by kdoh** [comment]
@mazameli Yeah that's in the ToDo list already.

**Comment by mazameli** [comment]
Feels like the default space between columns could be tightened up a bit. I'm seeing situations like this pretty often:

screen shot 2018-10-11 at 1 02 38 pm

**Comment by tlrobinson** [comment]
Maybe flip the bubbles' background to white if the cell has a bg color?

**Comment by tlrobinson** [comment]
@kdoh do we want the bubbles in TableSimple (used in dashboards and chart settings) too?

File: frontend/src/metabase/visualizations/components/TableInteractive.jsx [hunk 1]
```diff
@@ -345,12 +345,15 @@ export default class TableInteractive extends Component {
...
         }}
-        className={cx("TableInteractive-cellWrapper", {
+        className={cx("TableInteractive-cellWrapper text-dark", {
           "TableInteractive-cellWrapper--firstColumn": columnIndex === 0,
           "TableInteractive-cellWrapper--lastColumn":
             columnIndex === cols.length - 1,
+          "TableInteractive-emptyCell": value == null,
           "cursor-pointer": isClickable,
           "justify-end": isColumnRightAligned(column),
+          "TableInteractive--PK": isID(column),
```
Comment (tlrobinson) [inline frontend/src/metabase/visualizations/components/TableInteractive.jsx]: Should this be `"TableInteractive--PK": isPK(column),`?

**Comment by kdoh** [comment]
@tlrobinson - @mazameli and I were hoping we could get the two to look as similar as possible in terms of visual style and behavior. I can't remember exactly why we split the two out other than the pagination feature. Anyhoo anything we can do to get them to behave and look similar is 👍

File: frontend/src/metabase/visualizations/components/TableInteractive.jsx (hunk 1 above)
Comment (tlrobinson) [reply frontend/src/metabase/visualizations/components/TableInteractive.jsx]: Or `"TableInteractive--ID": isID(column),`

Comment (kdoh) [reply frontend/src/metabase/visualizations/components/TableInteractive.jsx]: Yeah was just about to suggest the `TableInteractive--ID` solution 👍

**Comment by mazameli** [comment]
>  I can't remember exactly why we split the two out other than the pagination feature.

I think part of it was also in order to disable the column heading actions menu because there was a feeling that the actions other than the sorting options felt out of place in a dashboard context.

**Comment by tlrobinson** [comment]
The main difference is it's paginated instead of virtualized. `TableInteractive` also allows column reordering/resizing while `TableSimple` just relies on an HTML `table` for column sizing.

**Review by tlrobinson — APPROVED** [review]

**Pull request merged**